analytic_engine/src/sst/parquet/meta_data.rs

```diff
@@ -45,89 +58,201 @@ pub enum Error {
 
 define_result!(Error);
 
-const DEFAULT_BLOOM_FILTER_VERSION: u32 = 0;
+const DEFAULT_FILTER_VERSION: u32 = 0;
 
-#[derive(Debug, Clone, PartialEq, Eq, Default)]
-pub struct RowGroupBloomFilter {
-    // The column filter can be None if the column is not indexed.
-    column_filters: Vec<Option<Bloom>>,
+/// Filter can be used to test whether an element is a member of a set.
+/// False positive matches are possible if space-efficient probabilistic data
+/// structure are used.
+trait Filter: fmt::Debug {
+    /// Check the key is in the bitmap index.
+    fn contains(&self, key: &[u8]) -> bool;
+
+    /// Serialize the bitmap index to binary array.
+    fn to_bytes(&self) -> Vec<u8>;
+
+    /// Deserialize the binary array to bitmap index.
+    fn from_bytes(buf: Vec<u8>) -> Result<Self>
+    where
+        Self: Sized;
+}
+
+/// Filter based on https://docs.rs/xorfilter-rs/latest/xorfilter/struct.Xor8.html
+#[derive(Default)]
+struct Xor8Filter {
+    xor8: Xor8,
+}
+
+impl fmt::Debug for Xor8Filter {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str("XorFilter")
+    }
+}
+
+impl Filter for Xor8Filter {
```

Comment:
How about `ColumnFilter`?

Reply:
It's a general filter, not designed for column, so I prefer current name.